components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/CmdUtils.java

```diff
@@ -820,6 +926,79 @@ public static void createDeploymentConfig(String projectName, String deploymentC
         }
     }
 
+    /**
+     * Download file
+     *
+     * @param source source URL
+     * @param destination destination path to save file
+     * @param headers api definition download request headers
+     * @param values api definition download values for request headers
+     */
+    private static String downloadFile(String source, String destination, ArrayList<String> headers,
+                                     ArrayList<String> values, boolean insecure) throws IOException {
+
+        URL url;
+        if (insecure) {
+            useInsecureSSL();
+        }
+        HttpURLConnection urlConn = null;
+        FileOutputStream outputStream = null;
+        String saveFilePath;
+        try {
```

Comment:
Use `try-with-resource` to close the connection and outpuStream

Reply:
Cannot do as there is the logic inside.